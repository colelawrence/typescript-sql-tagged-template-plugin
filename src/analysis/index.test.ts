import { readdirSync, readFileSync } from "fs";
import { join as joinPath } from "path";
import { analyze } from ".";

describe("analyze snapshots", () => {
	const testcaseDir = joinPath(__dirname, "__testcases__");
	const files = readdirSync(testcaseDir);
	for (const file of files) {
		const sql = readFileSync(joinPath(testcaseDir, file), "utf8");
		test(file, () => {
			const result = analyze(sql);
			expect(result).toMatchSnapshot();
		});
	}
});

describe(analyze, () => {
	test("parameter location", () => {
		const sql =
			"SELECT * FROM orders WHERE user_id = $1 AND status = ANY($2)";
		const result = analyze(sql);
		expect(result.parameters.length).toBe(2);

		const p1 = result.parameters[0];
		const p1Text = "$" + p1.index;
		const p1TextFromLoc = sql.substr(p1.location, p1Text.length);
		expect(p1TextFromLoc).toBe(p1Text);

		const p2 = result.parameters[1];
		const p2Text = "$" + p2.index;
		const p2TextFromLoc = sql.substr(p2.location, p2Text.length);
		expect(p2TextFromLoc).toBe(p2Text);
	});
});
