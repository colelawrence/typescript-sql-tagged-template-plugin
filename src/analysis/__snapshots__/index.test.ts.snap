// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`any.sql 1`] = `
Object {
  "parameters": Map {
    1 => Object {
      "column": "status",
      "isArray": true,
      "schema": undefined,
      "table": "orders",
    },
  },
  "returnValues": Array [
    Object {
      "alias": undefined,
      "schema": undefined,
      "table": "orders",
      "type": "star",
    },
  ],
  "warnings": Array [],
}
`;

exports[`json-operator.sql 1`] = `
Object {
  "parameters": Map {
    1 => Object {
      "column": "shipment_data",
      "isArray": false,
      "jsonPath": Object {
        "isText": true,
        "path": "company",
      },
      "schema": undefined,
      "table": "orders",
    },
  },
  "returnValues": Array [
    Object {
      "alias": undefined,
      "column": "order_id",
      "schema": undefined,
      "table": "orders",
      "type": "column",
    },
    Object {
      "alias": undefined,
      "column": "notes",
      "schema": undefined,
      "table": "orders",
      "type": "column",
    },
    Object {
      "alias": undefined,
      "column": "status",
      "schema": undefined,
      "table": "orders",
      "type": "column",
    },
  ],
  "warnings": Array [],
}
`;

exports[`select.sql 1`] = `
Object {
  "parameters": Map {},
  "returnValues": Array [
    Object {
      "alias": undefined,
      "constType": "integer",
      "type": "const",
      "value": 1,
    },
  ],
  "warnings": Array [],
}
`;
